survival/server/structs.go

```diff
@@ -0,0 +1,139 @@
+package survivalserver
```

Comment:
This file seems to be containing a bit of everything. This probably should be better organized/documented/declared elsewhere, as we cannot understand from looking at this file what this is all about.